src/regmod/variable.py

```diff
@@ -82,29 +184,112 @@ def rm_priors(self, indices: Union[int, List[int], List[bool]]):
         self.process_priors()
 
     def get_mat(self, data: Data) -> np.ndarray:
+        """Get design matrix.
+
+        Parameters
+        ----------
+        data : Data
+            Data object that provides the covariates.
+
+        Returns
+        -------
+        np.ndarray
+            Design matrix.
+        """
         self.check_data(data)
         return data.get_covs(self.name)
 
     def get_gvec(self) -> np.ndarray:
+        """Get direct Gaussian prior vector.
+
+        Returns
+        -------
+        np.ndarray
+            Direct Gaussian prior vector.
+        """
         if self.gprior is None:
             gvec = np.repeat([[0.0], [np.inf]], self.size, axis=1)
         else:
             gvec = np.vstack([self.gprior.mean, self.gprior.sd])
         return gvec
 
     def get_uvec(self) -> np.ndarray:
+        """Get direct Uniform prior vector.
+
+        Returns
+        -------
+        np.ndarray
+            Direct Uniform prior vector.
+        """
         if self.uprior is None:
             uvec = np.repeat([[-np.inf], [np.inf]], self.size, axis=1)
         else:
             uvec = np.vstack([self.uprior.lb, self.uprior.ub])
         return uvec
 
     def copy(self) -> "Variable":
+        """Copy current instance.
+
+        Returns
+        -------
+        Variable
+            Current instance.
+        """
         return deepcopy(self)
 
 
 @dataclass
 class SplineVariable(Variable):
+    """Spline variable that store information of variable with splines.
+
+    Parameters
+    ----------
+    spline : XSpline, optional
+        Spline object that in charge of creating design matrix. Default to be
+        `None`. `spline` and `spline_specs` cannot be `None` at the same time.
+    spline_specs : SplineSpecs, optional
+        Spline settings used to create spline object. Recommend to use only when
+        use `knots_type={'rel_domain', 'rel_freq'}. Default to be `None`.
+    linear_gpriors : List[LinearPrior], optional
+        A list of linear Gaussian priors usually for shape priors of the spline.
+        Default to be an empty list.
+    linear_upriors : List[LinearPrior], optional
+        A list of linear Uniform priors usually for shape priors of the spline.
+        spline. Default to be an empty list.
```

Comment:
Unlike `gprior` and `uprior`, why do we want `linear_gpriors` and `linear_upriors` to be lists? I mean, is there an instance that we would specify multiple priors?

Reply:
Yes! There are indeed cases where we want to have multiple priors. For example, we might want to control the spline to be monotonic increasing and the value of the spline cannot go above 1, then we will need to have two spline priors for that.

Reply:
I see. Previously, we just need to set `monotonic=True`, but here since the priors need to be Gaussian or Uniform, can you show how to specify it?

Reply:
Yes you will need to add a `SplineUniformPrior(order=1, lb=0.0, ub=np.inf)` in the prior list.